components/director/internal/open_resource_discovery/service.go

```diff
@@ -768,40 +697,68 @@ func (s *Service) processWebhooksAndDocuments(ctx context.Context, tx persistenc
 			log.C(ctx).WithError(err).Errorf("error processing ORD documents: %v", err)
 		} else {
 			log.C(ctx).Info("Successfully processed ORD documents")
-			return true, tx.Commit()
+			return tx.Commit()
 		}
 	}
-	return false, nil
+	return nil
 }
 
-func (s *Service) getAppTemplatesWebhooks(ctx context.Context) (map[string][]*model.Webhook, error) {
+func (s *Service) getWebhooksWithOrdType(ctx context.Context) ([]*model.Webhook, error) {
 	tx, err := s.transact.Begin()
 	if err != nil {
 		return nil, err
 	}
 	defer s.transact.RollbackUnlessCommitted(ctx, tx)
+
 	ctx = persistence.SaveToContext(ctx, tx)
+	ordWebhooks, err := s.webhookSvc.ListByWebhookTypeWithSelectForUpdate(ctx, model.WebhookTypeOpenResourceDiscovery)
```

Comment:
If we are not updating the webhook entity, probably we should not use select for update? What do you think

Reply:
i agree with that one